tests/plugins/test_double_plugin.py

```diff
@@ -0,0 +1,42 @@
+import torch
+
+from pytorch_lightning import Trainer
+from tests.helpers import BoringModel
+
+
+class DoublePrecisionBoringModel(BoringModel):
+
+    def training_step(self, batch, batch_idx):
+        output = self((batch, torch.ones_like(batch).long()))  # Add some non floating-point data
```

Comment:
can you also assert that here the data is of fp32? Just to make sure that we really convert and not our tests just pass because we passed already converted data :)